lib/spack/spack/spec.py

```diff
@@ -523,7 +616,7 @@ def virtual(self):
     @staticmethod
     def is_virtual(name):
         """Test if a name is virtual without requiring a Spec."""
-        return not spack.repo.exists(name)
+        return name != '' and not spack.repo.exists(name)
```

Comment:
I don't like it that `name` is the only `Spec` field with different semantics from the others.  The others are `None` when undefined, but `name` is `''`.  I think that's confusing for future maintainers of `Spec`.  Can you change it so that `name` is `None` for nameless specs?